### DISCUSSION TITLE
How I can change locale when I use no_prefix strategy?

### DISCUSSION THREAD

**Discussion by zozulinskyi** [nuxt-modules/i18n#1826] [Q&A] [answered]
Hello everyone. I like a `strategy: no_prefix`
But I can't change locale when I use next code:
```
const switchLocalePath = useSwitchLocalePath();
....

<nuxt-link :to="switchLocalePath('en')">English</nuxt-link>
```

Any ideas how I can fix that?
Nuxt: 3.1.0
Nuxtjs/i18n: 8.0.0-beta.8

**Comment by ineshbose** [comment]
Would you be able to provide a reproduction on Stackblitz to help understand why your locale isn't changing in `no_prefix` strategy? On my testing, it appears to be working fine!

**Reply by zozulinskyi** [reply]
Yeah, sure.
https://stackblitz.com/edit/nuxt3-jmqbek?file=README.md

**Reply by ineshbose** [reply]
Thanks! Good to know `setLocale` is working; I've opened an issue meanwhile.

**Comment by kazupon** [comment, marked as answer]
@zozulinskyi 
Hi!

The `no_prefix` strategy does not allow you to use the function returned by `useSwitchLocalePath`, i.e. `switchLocalePath`, to switch languages. As the docs say, you need to use `setLocale` to switch languages.
https://v8.i18n.nuxtjs.org/guide/routing-strategies#no_prefix

The behavior of `switchLocalePath` works the same behavior since nuxt i18n module v7.

`switchLocalePath` works if you use the `no_prefix` strategy and routing using `pages`.

**Reply by kazupon** [reply]
I think elk might be a good reference for the nuxt3 i18n project using the `no_prefix` starategy.
https://github.com/elk-zone/elk

As far as I know, Tha’s a large project using nuxt i18n module in nuxt3.

I believe this will be your reference

**Reply by zozulinskyi** [reply]
@kazupon Thank you for answer.
